File: puzzles/99_Bottles/Rust/99_Bottles.rs
// Sings '99 Bottles of Beer'
fn main()
{
    for bottle in (1..100).rev()
    {
        if bottle > 1
        {
            println!("{} bottles of beer on the wall, {} bottles of beer.", bottle, bottle);
            if bottle > 2
            {
                println!("Take one down and pass it around, {} bottles of beer on the wall.", bottle - 1);
            }
            else
            {
                println!("Take one down and pass it around, 1 bottle of beer on the wall.");
            }
        }
        else
        {
            println!("1 bottle of beer on the wall, 1 bottle of beer.");
            println!("Take one down and pass it around, no more bottles of beer on the wall.");
        }
        println!("");
    }
    println!("No more bottles of beer on the wall, no more bottles of beer.");
    println!("Go to the store and buy some more, 99 bottles of beer on the wall.\n");
}

File: puzzles/99_Bottles/javascript/99bottles.js
let bottles = 99;
while (bottles != 0)
{
	if (bottles === 1) {
		console.log(`${bottles} bottle of beer on the wall, ${bottles} bottles of beer.`);
	} else {
		console.log(`${bottles} bottles of beer on the wall, ${bottles} bottles of beer.`);
	}
	
	--bottles;
	
	if (bottles === 1) {
		console.log(`Take one down and pass it around, ${bottles} bottle of beer on the wall.`);
	} else if (bottles === 0) {
		console.log(`Take one down and pass it around, no more bottles of beer on the wall.`);
	} else {
		console.log(`Take one down and pass it around, ${bottles} bottles of beer on the wall.`);
	}
}
console.log('No more bottles of beer on the wall, no more bottles of beer.');
console.log('Go to the store and buy some more, 99 bottles of beer on the wall.');
